internal/testrunner/runners/system/servicedeployer/kubernetes.go

```diff
@@ -0,0 +1,269 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package servicedeployer
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/install"
+	"github.com/elastic/elastic-package/internal/logger"
+	"github.com/elastic/elastic-package/internal/stack"
+)
+
+const (
+	kindControlPlaneContainerName = "kind-control-plane"
+
+	kindContext = "kind-kind"
+)
+
+// KubernetesServiceDeployer is responsible for deploying resources in the Kubernetes cluster.
+type KubernetesServiceDeployer struct {
+	definitionsDir string
+}
+
+type kubernetesDeployedService struct {
+	ctxt ServiceContext
+
+	definitionsDir string
+}
+
+func (s kubernetesDeployedService) TearDown() error {
+	logger.Debugf("uninstall custom Kubernetes definitions (directory: %s)", s.definitionsDir)
+
+	definitionPaths, err := findKubernetesDefinitions(s.definitionsDir)
+	if err != nil {
+		return errors.Wrapf(err, "can't find Kubernetes definitions in given directory (path: %s)", s.definitionsDir)
+	}
+
+	if len(definitionPaths) == 0 {
+		logger.Debugf("no custom definitions found (directory: %s). Nothing will be uninstalled.", s.definitionsDir)
+		return nil
+	}
+
+	err = modifyKubernetesResources(false, definitionPaths...)
+	if err != nil {
+		return errors.Wrapf(err, "can't uninstall Kubernetes resources (path: %s)", s.definitionsDir)
+	}
+	return nil
+}
+
+func (s kubernetesDeployedService) Signal(signal string) error {
+	return errors.New("signal is not supported")
+}
+
+func (s kubernetesDeployedService) Context() ServiceContext {
+	return s.ctxt
+}
+
+func (s kubernetesDeployedService) SetContext(sc ServiceContext) error {
+	s.ctxt = sc
+	return nil
+}
+
+var _ DeployedService = new(kubernetesDeployedService)
```

Comment:
Maybe I'm missing something obvious or some idiom, but why do we need this line?

Reply:
https://golang.org/doc/faq#guarantee_satisfies_interface

Reply:
Cool, I was not aware of this. Thanks!